### PR TITLE
vdk-jupyter: fix bug for failed requests and improve error handling

### PR DESCRIPTION
**Pull request by yonitoo**
What: Bug fix. Since the Task Runner change got in, if you tried to deploy a job without keytab, you would not have been informed about the error in any way and the Task Runner state wouldn't have been cleaned, therefore, not being able to handle any other request. Only a restart of the session would have solved this which is not good. 
To address this I:
* Modified the Task Runner start_task() method to proceed after 'failed' last request.
* Improved the error handling - show the error of failing request which was not shown and made the backend handlers able to detect any errors.

Why: To make it work in a better way and not to have to restart the session on each bad request.

Testing Done: Introduced relevant tests. The rest of the tests are passing.

### CODE REVIEW

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/serverRequests.ts [hunk 1]
```diff
@@ -104,11 +104,13 @@ const pollForTaskCompletion = async (
         `taskStatus?taskId=${taskId}`,
         { method: 'GET' }
       );
-      if (
-        result.task_id === taskId &&
-        (result.status === 'completed' || result.error)
-      ) {
-        return result;
+      if (result.task_id === taskId) {
+        if (result.status !== 'running') {
```
Comment (DeltaMichael) [inline projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/serverRequests.ts:108]: Why check this if you're going to check `result.status === 'failed'` after?

**Review by DeltaMichael — APPROVED** [review]

File: projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/serverRequests.ts (hunk 1 above)
Comment (yonitoo) [reply projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/serverRequests.ts:108]: 3 scenarios for status:
running - continue;
completed - return result;
failed - show error and return result.

**Review by antoniivanov — COMMENTED** [review]
I don't really see tests covering the scenario

**Review by gageorgiev — APPROVED** [review]
looks good, add a test or two and go ahead

**Pull request merged**